### PR TITLE
Update Leptonica to 1.73

### PR DESCRIPTION
**Pull request by Shreeshrii**

### CODE REVIEW

**Comment by Alexpux** [comment]
Update checksums too

File: mingw-w64-leptonica/PKGBUILD [hunk 1]
```diff
@@ -19,31 +19,32 @@ depends=(${MINGW_PACKAGE_PREFIX}-gcc-libs
...
+  cd "${srcdir}/build-${CARCH}"
   ../${_realname}-${pkgver}/configure \
+  -v --disable-static --disable-dependency-tracking --disable-silent-rules CFLAGS='-g -O2 -DMINIMUM_SEVERITY=L_SEVERITY_WARNING -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable -Wno-maybe-uninitialized' \
     --build=${MINGW_CHOST} \
     --host=${MINGW_CHOST} \
     --target=${MINGW_CHOST} \
     --prefix=${MINGW_PREFIX}
 
-  make
+  make LDFLAGS='-Wc,-static-libgcc'
```
Comment (Alexpux) [inline mingw-w64-leptonica/PKGBUILD]: Why it here?
We build shared as preferred version.

File: mingw-w64-leptonica/PKGBUILD [hunk 2]
```diff
@@ -19,31 +19,32 @@ depends=(${MINGW_PACKAGE_PREFIX}-gcc-libs
...
 
 build() {
-  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
-  mkdir "${srcdir}/build-${MINGW_CHOST}"
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  [[ -d "${srcdir}/build-${CARCH}" ]] && rm -rf "${srcdir}/build-${CARCH}"
+  mkdir "${srcdir}/build-${CARCH}"
+  cd "${srcdir}/build-${CARCH}"
   ../${_realname}-${pkgver}/configure \
+  -v --disable-static --disable-dependency-tracking --disable-silent-rules CFLAGS='-g -O2 -DMINIMUM_SEVERITY=L_SEVERITY_WARNING -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable -Wno-maybe-uninitialized' \
```
Comment (Alexpux) [inline mingw-w64-leptonica/PKGBUILD]: Split command arguments one by line.

File: mingw-w64-leptonica/PKGBUILD (hunk 1 above)
Comment (Shreeshrii) [inline mingw-w64-leptonica/PKGBUILD]: Notes from an msys installation:

​
>
   1. Configuration requires prior installation of the GIF, JPEG, PNG,
> TIFF, and

      WEBP libraries if I/O corresponding to these formats is to be
> supported,

      and of the ZLIB library if compression is to be used.


>    2. "--disable-static" suppresses building the static library.


>    3. "--disable-silent-rules" is required to show compiler invocations.


>    4. "-DMINIMUM_SEVERITY=L_SEVERITY_WARNING" suppresses informational
> messages

      in the library, i.e., allows only warning and error messages to be

      printed.


>    5. The compiler warning options are a sanity check on the library code.


>    6. "-Wc,-static-libgcc" prevents libtool from linking the gcc runtime
> DLL to

      the library DLL.  It must be specified during building, as it is a
> libtool

      (not gcc) option; specifying it during configuration will fail.


>    7. The default compiler flags are "-g -O2", and the default linker
> flags are

      "-Wl,--as-needed".


>    8. "-C src" skips installation of 13 utilities from the "prog"
> subdirectory.

​
>
>
ShreeDevi
____________________________________________________________
भजन - कीर्तन - आरती @ http://bhajans.ramparivar.com

Comment (Alexpux) [inline mingw-w64-leptonica/PKGBUILD]: We link with libgcc runtime. So don't use --static-libgcc

File: mingw-w64-leptonica/PKGBUILD (hunk 2 above)
Comment (Alexpux) [inline mingw-w64-leptonica/PKGBUILD]: Do not pass your CFLAGS to configure. Before configure write:
````CFLAGS+=" -DMINIMUM_SEVERITY=L_SEVERITY_WARNING"````

File: mingw-w64-leptonica/PKGBUILD [hunk 3]
```diff
@@ -19,31 +19,32 @@ depends=(${MINGW_PACKAGE_PREFIX}-gcc-libs
...
+#  cd "${srcdir}/build-${CARCH}"
+#  make -j1 check
+#}
 
 package() {
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  cd "${srcdir}/build-${CARCH}"
   make DESTDIR="${pkgdir}" install
-}
+  }
```
Comment (Alexpux) [inline mingw-w64-leptonica/PKGBUILD]: There are no spaces need before "}"

File: mingw-w64-leptonica/PKGBUILD [hunk 4]
```diff
@@ -19,31 +19,35 @@ depends=(${MINGW_PACKAGE_PREFIX}-gcc-libs
...
 }
 
 build() {
-  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
-  mkdir "${srcdir}/build-${MINGW_CHOST}"
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  [[ -d "${srcdir}/build-${CARCH}" ]] && rm -rf "${srcdir}/build-${CARCH}"
+  mkdir "${srcdir}/build-${CARCH}"
+  cd "${srcdir}/build-${CARCH}"
+  CFLAGS='-g -O2 -DMINIMUM_SEVERITY=L_SEVERITY_WARNING -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable -Wno-maybe-uninitialized'
```
Comment (Alexpux) [inline mingw-w64-leptonica/PKGBUILD]: Do as I wrote. Don't override ALL CFLAGS. You need only add "-DMINIMUM_SEVERITY=L_SEVERITY_WARNING" to current CFLAGS.

File: mingw-w64-leptonica/PKGBUILD [hunk 5]
```diff
@@ -19,31 +19,35 @@ depends=(${MINGW_PACKAGE_PREFIX}-gcc-libs
...
-  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
-  mkdir "${srcdir}/build-${MINGW_CHOST}"
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  [[ -d "${srcdir}/build-${CARCH}" ]] && rm -rf "${srcdir}/build-${CARCH}"
+  mkdir "${srcdir}/build-${CARCH}"
+  cd "${srcdir}/build-${CARCH}"
+  CFLAGS='-g -O2 -DMINIMUM_SEVERITY=L_SEVERITY_WARNING -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable -Wno-maybe-uninitialized' 
   ../${_realname}-${pkgver}/configure \
+    -v \
+    --disable-static \
```
Comment (Alexpux) [inline mingw-w64-leptonica/PKGBUILD]: Remove this. We build both static and shared library

File: mingw-w64-leptonica/PKGBUILD [hunk 6]
```diff
@@ -19,31 +19,35 @@ depends=(${MINGW_PACKAGE_PREFIX}-gcc-libs
...
+    --disable-dependency-tracking \
+    --disable-silent-rules \
     --build=${MINGW_CHOST} \
     --host=${MINGW_CHOST} \
     --target=${MINGW_CHOST} \
     --prefix=${MINGW_PREFIX}
-
-  make
+  make LDFLAGS='-Wc'
```
Comment (Alexpux) [inline mingw-w64-leptonica/PKGBUILD]: Just "make", nothing more

**Comment by Shreeshrii** [comment]
$ tesseract  --tessdata-dir /mingw32/share ./testing/toc2.gif ./testting/toc2
Tesseract Open Source OCR Engine v3.05.00dev with Leptonica
Warning in pixReadMemGif: writing to a temp file, not directly to memory
Error in pixReadStreamGif: Can't use giflib-5.1.2; suggest 5.1.1 or earlier

need to change depends

         '${MINGW_PACKAGE_PREFIX}-giflib=5.1.1'

**Comment by Alexpux** [comment]
Well need to fix tesseract to use current gif lib instead, because I don't want to downgrade giflib

**Comment by rockyfire** [comment]
[giflib-5.1.1 download](https://nchc.dl.sourceforge.net/project/giflib/giflib-5.1.1.tar.bz2)
@Alexpux 
but i don't know how to use it

**Pull request merged**